src/ceph-volume/ceph_volume/api/lvm.py

```diff
@@ -1290,3 +1306,45 @@ def create_lvs(volume_group, parts=None, size=None, name_prefix='ceph-lv'):
             create_lv(lv_name, volume_group.name, extents=extents, tags=tags)
         )
     return lvs
+
+
+#############################################################
+#
+# New methods to get PVs, LVs, and VGs.
+# Later, these can be easily merged with get_api_* methods
+#
+###########################################################
+
+PV_FIELDS = 'pv_name,pv_tags,pv_uuid,vg_name,lv_uuid'
+VG_FIELDS = 'vg_name,pv_count,lv_count,snap_count,vg_attr,vg_size,vg_free,vg_free_count'
+LV_FIELDS = 'lv_tags,lv_path,lv_name,vg_name,lv_uuid,lv_size'
+
+def get_pvs(fields=PV_FIELDS, sep='";"', filters=None):
+    args = ['pvs', '--no-heading', '--readonly', '--separator=' + sep,
+            '-o', fields]
+    if filters:
+        args.extend(['-S', filters])
+
+    stdout, stderr, returncode = process.call(args, verbose_on_failure=False)
+
+    return _output_parser(stdout, fields)
+
+def get_vgs(fields=VG_FIELDS, sep='";"', filters=None):
+    args = ['vgs', '--no-heading', '--readonly', '--separator=' + sep,
+            '-o', fields]
+    if filters:
+        args.extend(['-S', filters])
+
+    stdout, stderr, returncode = process.call(args, verbose_on_failure=False)
+
+    return _output_parser(stdout, fields)
+
+def get_lvs(fields=LV_FIELDS, sep='";"', filters=None):
+    args = ['lvs', '--no-heading', '--readonly', '--separator=' + sep,
+            '-o', fields]
+    if filters:
+        args.extend(['-S', filters])
+
+    stdout, stderr, returncode = process.call(args, verbose_on_failure=False)
+
+    return _output_parser(stdout, fields)
```

Comment:
Nope. I've tried to call the `lvs`/`pvs`/`vgs` command as less frequently as possible/tried to use cache as far as possible. I should've removed part since it doesn't belong to this PR.